ereolen_support/ereolen_support.admin.form.inc

```diff
@@ -0,0 +1,111 @@
+<?php
+
+/**
+ * @file
+ * Add eReolen support admin form.
+ */
+
+/**
+ * Create backend admin form.
+ *
+ * @param array $form
+ *   The form to create.
+ * @param array $form_state
+ *   The state of the form.
+ *
+ * @return mixed
+ */
+function ereolen_support_config_form($form, &$form_state) {
+  $fieldTypesBuild = ['select', 'textfield', 'textarea', 'emailfield', 'date_popup'];
+  $frontendForm = _ereolen_support_retrieve_form('ereolen_support_form');
+
+  $form['support_form'] = array (
+    '#type' => 'vertical_tabs',
+  );
+  $form['jira_settings'] = array(
+    '#type' => 'fieldset',
+    '#title' => t('Jira settings'),
+    '#group' => 'support_form',
+    '#collapsible' => FALSE,
+    '#access'=> user_access('administer ereolen support form')
+  );
+  $form['field_config'] = array(
+    '#type' => 'fieldset',
+    '#title' => t('Field config'),
+    '#group' => 'support_form',
+    '#collapsible' => FALSE,
+  );
+  $form['jira_settings']['jira_domain'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Jira domain'),
+    '#default_value' => variable_get('jira_domain'),
+  );
+
+  $form['jira_settings']['jira_username'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Jira username'),
+    '#default_value' => variable_get('jira_username'),
+  );
+
+  $form['jira_settings']['jira_password'] = array(
+    '#type' => 'password',
+    '#title' => t('Jira user password'),
+    '#default_value' => variable_get('jira_password'),
+  );
+
+  $form['jira_settings']['jira_project_id'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Jira project id'),
+    '#default_value' => variable_get('jira_project_id'),
+  );
+
+  $form['jira_settings']['jira_issue_type_id'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Jira issue type id'),
+    '#default_value' => variable_get('jira_issue_type_id'),
+  );
+
+  // Build field configuration from frontend form.
+  foreach ($frontendForm as $key => $possibleField) {
+    // Make sure the form entry is an array.
+    if(is_array($possibleField)) {
+      // Make sure it's a field type.
+      if(in_array('#type', $possibleField) && in_array($possibleField['#type'], $fieldTypesBuild)) {
```

Comment:
Can it not be on of these types and should an `drupal_set_message` or `watchdog` inform that an non-field-type was found?

Reply:
I'm not sure what the first part of the question is about. On the part about watchdog i don't believe a watchdog has much use in this instance.

Reply:
Watchdog was for us when they come with an jira support ticket to help track down an error.